src/main/java/mate/academy/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,51 @@
+package mate.academy.dao;
+
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't save Movie to DB", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return movie;
+    }
+
+    @Override
+    public Optional<Movie> get(Long id) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Optional<Movie> movie;
+        try {
+            session = sessionFactory.openSession();
+            movie = Optional.ofNullable(session.get(Movie.class, id));

```

Comment:
```suggestion
            return Optional.ofNullable(session.get(Movie.class, id));

```
Using this way, you can remove two redundant lines of code: `Optional<Movie> movie;` and `return movie;`